### PR TITLE
Bug: LT-2647 - oiel type application performance improvements

### PR DESCRIPTION
**Pull request by Mark-Spiers-kainos**

### CODE REVIEW

File: applications/serializers/advice.py [hunk 1]
```diff
@@ -154,9 +154,9 @@ def _footnote_fields_setup(self):
 
 
 class CountryWithFlagsSerializer(serializers.Serializer):
-    id = serializers.CharField()
-    name = serializers.CharField()
-    flags = serializers.SerializerMethodField()
+    id = serializers.CharField(read_only=True)
+    name = serializers.CharField(read_only=True)
+    flags = serializers.SerializerMethodField(read_only=True)
```
Comment (BenAndrew-Kainos) [inline applications/serializers/advice.py:159]: Read only makes very little to no impact btw

File: applications/serializers/open_application.py [hunk 2]
```diff
@@ -72,8 +72,10 @@ class Meta:
         )
 
     def get_goods_types(self, application):
-        goods_types = application.goods_type.all().prefetch_related("countries")
-        default_countries = Country.include_special_countries.filter(countries_on_application__application=application)
+        goods_types = application.goods_type.all().prefetch_related("countries", "countries__flags")
+        default_countries = list(
+            Country.include_special_countries.filter(countries_on_application__application_id=application.id)
```
Comment (BenAndrew-Kainos) [inline applications/serializers/open_application.py:77]: I don't think changing application to application_id makes any performance difference. Have you checked this?

File: applications/serializers/open_application.py [hunk 3]
```diff
@@ -275,9 +280,11 @@ class CountryOnApplicationViewSerializer(serializers.Serializer):
 
     def get_country(self, instance):
         if self.context.get("active_flags_only"):
-            return CountryWithFlagsSerializer(instance.country, context={"with_active_flags": True}).data
+            return CountryWithFlagsSerializer(
+                instance.country, context={"with_active_flags": True}, read_only=True
```
Comment (BenAndrew-Kainos) [inline applications/serializers/open_application.py:284]: once again read-only makes negligible difference

File: applications/serializers/open_application.py (hunk 2 above)
Comment (Mark-Spiers-kainos) [reply applications/serializers/open_application.py:77]: It doesn't, by default it compares the pk of the obects, I just like it for clarity in reading

**Pull request merged**